github/Secret.py

```diff
@@ -0,0 +1,138 @@
+        return self._updated_at.value
+
+    @property
+    def visibility(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._visibility)
+        return self._visibility.value
+
+    @property
+    def selected_repositories(self):
+        """
+        :calls: `GET {secret_url}/repositories <https://docs.github.com/en/rest/actions/secrets#list-selected-repositories-for-an-organization-secret>`_
+        :type: List of type Repository
+        """
+        self._completeIfNotSet(self._selected_repositories)
+        return self._selected_repositories.value
+
+    @property
+    def url(self):
+        """
+        :type: string
+        """
+        return self._url.value
+
+    def delete(self):
+        """
+        :calls: `DELETE {secret_url} <https://docs.github.com/en/rest/actions/secrets>`_
+        :rtype: None
+        """
+        headers, data = self._requester.requestJsonAndCheck("DELETE", self.url)
+
+    def add_repo(self, repo):
+        """
+        :calls: 'PUT {org_url}/actions/secrets/{secret_name} <https://docs.github.com/en/rest/actions/secrets#add-selected-repository-to-an-organization-secret>`_
+        :param repo: github.Repository.Repository
+        :rtype: bool
+        """
+        if self.visibility != "selected":
+            return False
+        self._requester.requestJsonAndCheck("PUT", f"{self.url}/repositories/{repo.id}")
+        self._selected_repositories.value.append(repo)
+        return True
+
+    def remove_repo(self, repo):
+        """
+        :calls: 'DELETE {org_url}/actions/secrets/{secret_name} <https://docs.github.com/en/rest/actions/secrets#add-selected-repository-to-an-organization-secret>`_
+        :param repo: github.Repository.Repository
+        :rtype: bool
+        """
+        if self.visibility != "selected":
+            return False
+        self._requester.requestJsonAndCheck("DELETE", f"{self.url}/repositories/{repo.id}")
+        self._selected_repositories.value.remove(repo)
+        return True
+
+    def _initAttributes(self):
+        self._name = github.GithubObject.NotSet
+        self._created_at = github.GithubObject.NotSet
+        self._updated_at = github.GithubObject.NotSet
+        self._visibility = github.GithubObject.NotSet
+        self._selected_repositories = github.GithubObject.NotSet
+        self._url = github.GithubObject.NotSet
+
+    def _useAttributes(self, attributes):
+        if "name" in attributes:
+            self._name = self._makeStringAttribute(attributes["name"])
+        if "created_at" in attributes:
+            self._created_at = self._makeDatetimeAttribute(attributes["created_at"])
+        if "updated_at" in attributes:
+            self._updated_at = self._makeDatetimeAttribute(attributes["updated_at"])
+        if "visibility" in attributes:
+            self._visibility = self._makeStringAttribute(attributes["visibility"])
+        if "selected_repositories_url" in attributes:
```

Comment:
This means we are fetching the list of selected repositories whenever we create a secret instance, no matter if it is used or not. We like to avoid unnecessary API calls.

Further, this list of repos is paginated, so this would fetch only the first 30 repos.

Ideally, `selected_repositories` returns a paginated list of the repos:

    @property
    def selected_repositories(self) -> PaginatedList[Repository]:
        return github.PaginatedList.PaginatedList(
            github.Repository.Repository,
            self._requester,
            self.selected_repositories_url,
            {},
            list_item="repositories",
        )